mpqp/execution/connection/key_connection.py

```diff
@@ -0,0 +1,132 @@
+from getpass import getpass
+from typing import Callable, Optional
+from termcolor import colored
+
+from mpqp.execution.connection.env_manager import get_env_variable, save_env_variable
+
+def config_key(key_name :str, configuration_name: str, test_connection:  Callable[[str], bool]):
+    """
+    Configure a key by setting the API token.
+
+    Returns:
+        tuple: A message indicating the result of the configuration and an empty list.
+    """
+    was_configured = get_env_variable(f"{configuration_name}_CONFIGURED") == "True"
+
+    if was_configured:
+        decision = input(
+            f"{configuration_name} key is already configured. Do you want to update it? [y/N]"
+        )
+        if decision.lower().strip() != "y":
+            return "Canceled.", []
+
+    token = getpass(f"Enter your {configuration_name} token (hidden): ")
+    if token == "":
+        print(colored("Empty credentials", "red"))
+        getpass("Press 'Enter' to continue")
+        return "", []
+    if test_connection(token):
+        save_env_variable(f"{key_name}", token)
+        save_env_variable(f"{configuration_name}_CONFIGURED", "True")
+        return f"{configuration_name} key correctly configured", []
+    else:
+        if not was_configured:
+            save_env_variable(f"{configuration_name}_CONFIGURED", "False")
+        getpass("Press 'Enter' to continue")
+        return "", []
+    
+
+def config_ionq_key():
```

Comment:
This file is supposed to have general key configurations, but then only deal with ionq and aqt ? Shouldn't we move all this in google_connection, since today it is related with Cirq ?